bitcoin/src/pow.rs

```diff
@@ -489,20 +499,29 @@ impl CompactTarget {
         }
         // Comments relate to the `pow.cpp` file from Core.
         // ref: <https://github.com/dogecoin/dogecoin/blob/51cbc1fd5d0d045dda2ad84f53572bbf524c6a8e/src/dogecoin.cpp>
-        let max_timespan = params.pow_target_timespan << 2;
-        let min_timespan = if height > 10000 { // Lines 57-66
-            params.pow_target_timespan >> 2
-        } else if height > 5000 {
-            params.pow_target_timespan >> 3
-        } else {
-            params.pow_target_timespan >> 4
-        };
-        let actual_timespan = timespan.clamp(min_timespan, max_timespan); // Lines 69-72 nModulatedTimespan
+        let retarget_timespan = params.pow_target_timespan(height); // Line 44
+        let mut modulated_timespan = timespan; // Lines 45-46
+        if params.digishield_activated(height) { // Lines 50-56
+            modulated_timespan = retarget_timespan + (modulated_timespan - retarget_timespan) / 8; // Line 53
```

Comment:
nit: binary shift operator are more efficient:
```suggestion
            modulated_timespan = retarget_timespan + (modulated_timespan - retarget_timespan) >> 3; // Line 53
```

Reply:
This would work only if `modulated_timespan - retarget_timespan` was always positive, which is not the case. Thankfully, I have 5 tests which fail if I use >> 3

Reply:
Also, I believe you meant: `retarget_timespan + ((modulated_timespan - retarget_timespan) >> 3)` which is not the same as `retarget_timespan + (modulated_timespan - retarget_timespan) >> 3` in rust.

Reply:
That was definitely not the best suggestion 😅 .  Thanks for trying it out and catching the multiple problems with it.

> This would work only if `modulated_timespan - retarget_timespan` was always positive, which is not the case.

Good catch, I tend to forget that we are dealing with `i64` and not `u64`.

>  I believe you meant: `retarget_timespan + ((modulated_timespan - retarget_timespan) >> 3)`

Good catch, I forgot that the [precedence](https://doc.rust-lang.org/reference/expressions.html#expression-precedence) of shift operators is not the same as the arithmetic ones.